src/shared/machine-id-setup.c

```diff
@@ -30,6 +30,8 @@
 #include "umask-util.h"
 #include "virt.h"
 
+static bool arg_machine_id_from_firmware = false;
```

Comment:
uh, what? no. global state is generally frowned on in our codebase, except for some very exceptional cases. this doesn't qualify...

just pass this via function parameters?